.github/workflows/ci.yml

```diff
@@ -10,62 +7,58 @@ on:
       - '**/*.md'
       - '**/*.yml'
       - '**/*.yaml'
-      - .github/**
+      - '.github/**'
   pull_request:
     branches: [main]
     paths-ignore:
       - '**/*.md'
       - '**/*.yml'
       - '**/*.yaml'
-      - .github/**
+      - '.github/**'
   workflow_dispatch:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
-
+    runs-on: [ubuntu-latest]
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
-          node-version: 20
-          cache: 'npm'
-
+          cache: npm
+          node-version-file: .nvmrc
       - name: Setup SAP RFC SDK
         uses: ./.github/actions/actions-setup-sap-rfc
         with:
-          github-token: ${{ secrets.GH_TOKEN }}
-          patch-level: '11'
-
-      - name: Install dependencies
+          token: ${{ secrets.GH_TOKEN }}
+      - name: Install Dependencies
         run: npm ci
-
       - name: Build
         run: npm run build
-
       - name: Test
         run: npm run test
         env:
           CI: true
-
   release:
     needs: build
     if: github.ref == 'refs/heads/main'
-    runs-on: ubuntu-latest
+    runs-on: [ubuntu-latest]
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-node@v4
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
         with:
-          node-version: 20
-          cache: 'npm'
-
+          cache: npm
+          node-version-file: .nvmrc
       - name: Setup SAP RFC SDK
         uses: ./.github/actions/actions-setup-sap-rfc
         with:
-          github-token: ${{ secrets.GH_TOKEN }}
-          patch-level: '11'
-
-      - run: npm ci
-      - run: npm run semantic-release
+          token: ${{ secrets.GH_TOKEN }}
+      - name: Install Dependencies
+        run: npm ci
+      - name: Semantic Release
+        run: npm run semantic-release
         env:
-          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Na secrets e em outras partes que referenciam essa env, encontrei somente o nome antigo: **GH_TOKEN** e não **GITHUB_TOKEN**, está certo?
Aqui deveria ser GH_TOKEN?

Reply:
@alvarolimajr o `GH_TOKEN` deve ser utilizado somente no _checkout_ do repo de _action_. as demais utilizações devem ocorrer neste _token_ temporário gerado e utilizado apenas durante a execução deste _pipeline_. se for trocado possivelmente teremos `401`/`403`, pois o novo _token_ gerado foi criado considerando menor privilégio, ou seja, `read-only` apenas no repo de _action_